bus/bus.go

```diff
@@ -590,17 +590,44 @@ func (b *bus) walletTransactionsHandler(jc jape.Context) {
 		return
 	}
 
+	// convertToTransaction converts wallet event data to a Transaction.
+	convertToTransaction := func(kind string, data wallet.EventData) (txn types.Transaction, ok bool) {
```

Comment:
Would it not be better to store the events as is rather than trying to convert everything to a transaction type? This would be a good time since we're already breaking compatibility elsewhere.

Reply:
Definitely ignore this `DRAFT` for a bit, instead of adding a new `gorm` type I decided to first convert all chain code to SQL and then add the event in raw SQL. To answer your question though, we will be storing the events more or less as-is, this convert function is to make sure the output of our API endpoint `/wallet/transactions` still serves the same data.